cmd/local-gadget/utils/utils.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+// WaitForEnd waits until the user-set timeout happened or SIGINT or SIGTERM
+// is sent to us
+func WaitForEnd(f *CommonFlags) {
+	timeoutChannel := make(chan os.Signal, 1)
+	signal.Notify(timeoutChannel, os.Interrupt)
+	if f.Timeout != 0 {
+		go func() {
+			time.Sleep(time.Duration(f.Timeout) * time.Second)
+			timeoutChannel <- os.Interrupt
+		}()
+	}
```

Comment:
Why do you also need to attach the signal to this channel?  

What about something like the following?

```suggestion
	timeoutChannel := make(chan struct{})
	if f.Timeout != 0 {
		go func() {
			time.Sleep(time.Duration(f.Timeout) * time.Second)
			close(timeoutChannel)
		}()
	}
```

Reply:
Thinking better, we can use [time.After](https://pkg.go.dev/time#After) to avoid creating the goroutine and handling it outselves. What do you think?

```suggestion
	timeoutChannel := make(<-chan time.Time)
	if f.Timeout != 0 {
		timeoutChannel = time.After(time.Duration(f.Timeout) * time.Second)
	}
```